docs/Serial.md

```diff
@@ -56,86 +56,179 @@ e.g. after configuring a port for GPS enable the GPS feature.
-| ---------- | --------- |
-| 0          | Auto      |
-| 1          | 9600      |
-| 2          | 19200     |
-| 3          | 38400     |
-| 4          | 57600     |
-| 5          | 115200    |
-| 6          | 230400    |
-| 7          | 250000    |
+The `serial` CLI command takes 6 arguments:
+```
+serial <port identifier> <port function> <msp baudrate> <sensor baudrate> <telemetry baudrate> <peripheral baudrate>
+```
 
+| Argument                       | Default | Source                   |
+| ------------------------------ | ------- | ------------------------ |
+| 1. Serial Port Identifier      |         | serialPortIdentifier_e   |
+| 2. Serial Port Function        |         | serialPortFunction_e     |
+| 3. Configuration/MSP baud rate | 115200  | baudrate_e               |
+| 4. Sensor Input baud rate      | 57600   | gpsbaudRate_e            |
+| 5. Telemetry baud rate         | AUTO    | baudRate_e               |
+| 6. Peripherals baud rate       | 115200  | baudrate_e               |
+
+### 1. Serial Port Identifier
+
+| Identifier                 | Value |
+| -------------------------- | ----- |
+| SERIAL_PORT_NONE           | -1    |
+| SERIAL_PORT_USART1         | 0     |
+| SERIAL_PORT_USART2         | 1     |
+| SERIAL_PORT_USART3         | 2     |
+| SERIAL_PORT_UART4          | 3     |
+| SERIAL_PORT_UART5          | 4     |
+| SERIAL_PORT_USART6         | 5     |
+| SERIAL_PORT_USART7         | 6     |
+| SERIAL_PORT_USART8         | 7     |
+| SERIAL_PORT_LPUART1        | 8     |
+| SERIAL_PORT_USB_VCP        | 20    |
+| SERIAL_PORT_SOFTSERIAL1    | 30    |
+| SERIAL_PORT_SOFTSERIAL2    | 31    |
+
+### 2. Serial Port Function
+
+ | Function                     | Value |
+ | ---------------------------- | ----- |
+ | FUNCTION_NONE                | 0     |
+ | FUNCTION_MSP                 | 1     |
+ | FUNCTION_GPS                 | 2     |
+ | FUNCTION_TELEMETRY_FRSKY_HUB | 4     |
+ | FUNCTION_TELEMETRY_HOTT      | 8     |
+ | FUNCTION_TELEMETRY_LTM       | 16    |
+ | FUNCTION_TELEMETRY_SMARTPORT | 32    |
+ | FUNCTION_RX_SERIAL           | 64    |
+ | FUNCTION_BLACKBOX            | 128   |
+ | FUNCTION_TELEMETRY_MAVLINK   | 512   |
+ | FUNCTION_ESC_SENSOR          | 1024  |
+ | FUNCTION_VTX_SMARTAUDIO      | 2048  |
+ | FUNCTION_TELEMETRY_IBUS      | 4096  |
+ | FUNCTION_VTX_TRAMP           | 8192  |
+ | FUNCTION_RCDEVICE            | 16384 |
+ | FUNCTION_LIDAR_TF            | 32768 |
+ | FUNCTION_FRSKY_OSD           | 65536 |
+
+Note: `FUNCTION_FRSKY_OSD` = `(1<<16)` requires 17 bits.
+
+### 3. Configuration/MSP Baudrates
+
+| Baudrate |
+| -------- |
+| 9600     |
+| 19200    |
+| 38400    |
+| 57600    |
+| 115200   |
+| 230400   |
+| 250000   |
+| 500000   |
+| 1000000  |
+
+### 4 Sensor Input Baudrates
+
+Configure GPS or ESC Baud Rate according to device documentation.
+
+#### 4.1 GPS Baudrates
+
+| Baudrate ID         | Value |
+| ------------------- | ----- |
+| GPS_BAUDRATE_115200 | 0     |
```

Comment:
This seems to be backwards to what is done in the `serial` CLI command - GPS uses the same baud rate indices as everything else.

Reply:
I had used the code found in gps.c and gps.h. Changed to use common indices.